collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_info.py

```diff
@@ -585,9 +692,63 @@ def generate_snap_dict(blade):
             'suffix': snaps.items[snap].suffix,
             'source_destroyed': snaps.items[snap].source_destroyed,
         }
+        if REPLICATION_API_VERSION in api_version:
+            snap_info[snapshot]['owner'] = snaps.items[snap].owner.name
+            snap_info[snapshot]['owner_destroyed'] = snaps.items[snap].owner_destroyed
+            snap_info[snapshot]['source_display_name'] = snaps.items[snap].source_display_name
+            snap_info[snapshot]['source_is_local'] = snaps.items[snap].source_is_local
+            snap_info[snapshot]['source_location'] = snaps.items[snap].source_location.name
     return snap_info
 
 
+def generate_snap_transfer_dict(blade):
+    snap_transfer_info = {}
+    snap_transfers = blade.file_system_snapshots.list_file_system_snapshots_transfer()
+    for snap_transfer in range(0, len(snap_transfers.items)):
+        transfer = snap_transfers.items[snap_transfer].name
+        snap_transfer_info[transfer] = {
+            'completed': snap_transfers.items[snap_transfer].completed,
+            'data_transferred': snap_transfers.items[snap_transfer].data_transferred,
+            'progress': snap_transfers.items[snap_transfer].progress,
+            'direction': snap_transfers.items[snap_transfer].direction,
+            'remote': snap_transfers.items[snap_transfer].remote.name,
+            'remote_snapshot': snap_transfers.items[snap_transfer].remote_snapshot.name,
+            'started': snap_transfers.items[snap_transfer].started,
+            'status': snap_transfers.items[snap_transfer].status
+        }
+    return snap_transfer_info
+
+
+def generate_array_conn_dict(blade):
+    array_conn_info = {}
+    arrays = blade.array_connections.list_array_connections()
+    for arraycnt in range(0, len(arrays.items)):
+        array = arrays.items[arraycnt].remote.name
+        array_conn_info[array] = {
+            'encrypted': arrays.items[arraycnt].encrypted,
+            'replication_addresses': arrays.items[arraycnt].replication_addresses,
+            'management_address': arrays.items[arraycnt].management_address,
+            'id': arrays.items[arraycnt].remote.id,
+            'status': arrays.items[arraycnt].status,
+            'version': arrays.items[arraycnt].version,
+        }
+        if arrays.items[arraycnt].encrypted:
+            array_conn_info[array].ca_certificate_group = arrays.items[arraycnt].ca_certificate_group
+    return array_conn_info
+
+
+def generate_policies_dict(blade):
+    policies_info = {}
+    policies = blade.policies.list_policies()
+    for policycnt in range(0, len(policies.items)):
+        policy = policies.items[policycnt].name
+        policies_info[policy] = {
+            'enabled': policies.items[policycnt].enabled,
+            'rules': policies.items[policycnt].rules[0].to_dict()
```

Comment:
You should first check to see if `polices` has any rules, before referencing `rules[0]`. If there is a policy on the FB with no rules (which is possible), the module will fail with error `purefb_info.py, line 747, in generate_policies_dict\nIndexError: list index out of range`.

Reply:
Good catch - fixed